homeassistant/components/aprilaire/sensor.py

```diff
@@ -0,0 +1,309 @@
+            return True
+
+        if not super().available:
+            return False
+
+        return (
+            self.coordinator.data.get(self.entity_description.status_key)
+            == self.status_sensor_available_value
+        )
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the value reported by the sensor."""
+
+        # Valid cast as pyaprilaire only provides str | int | float
+        return cast(
+            StateType, self.coordinator.data.get(self.entity_description.value_key)
+        )
+
+
+class AprilaireHumiditySensor(BaseAprilaireSensor):
+    """Humidity sensor entity for Aprilaire."""
+
+    status_sensor_available_value = 0
+    status_sensor_exists_values = [0, 1, 2]
+
+
+class AprilaireTemperatureSensor(BaseAprilaireSensor):
+    """Temperature sensor entity for Aprilaire."""
+
+    status_sensor_available_value = 0
+    status_sensor_exists_values = [0, 1, 2]
+
+    @property
+    def suggested_display_precision(self) -> int | None:
+        """Return the suggested number of decimal digits for display."""
+        if self.unit_of_measurement == UnitOfTemperature.CELSIUS:
+            return 1
+
+        return 0
+
+
+class AprilaireStatusSensor(BaseAprilaireSensor):
+    """Status sensor entity for Aprilaire."""
+
+    status_sensor_exists_values = [1, 2]
+    entity_description: AprilaireStatusSensorDescription
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the value reported by the sensor mapped to the status option."""
+
+        raw_value = super().native_value
+
+        return self.entity_description.status_map.get(cast(int, raw_value))
+
+
+def get_entities(
+    entity_class: type[BaseAprilaireSensor],
+    coordinator: AprilaireCoordinator,
+    unique_id: str,
+    descriptions: tuple[AprilaireSensorDescription, ...],
+) -> list[BaseAprilaireSensor]:
+    """Get the entities for a list of sensor descriptions."""
+
+    entities = (
+        entity_class(coordinator, description, unique_id)
+        for description in descriptions
+    )
+
+    return [entity for entity in entities if entity.exists]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Aprilaire sensor devices."""
+
+    coordinator: AprilaireCoordinator = hass.data[DOMAIN][config_entry.unique_id]
+
+    assert config_entry.unique_id is not None
+
+    entities = (
+        get_entities(
+            AprilaireHumiditySensor,
+            coordinator,
+            config_entry.unique_id,
+            HUMIDITY_SENSORS,
+        )
+        + get_entities(
+            AprilaireTemperatureSensor,
+            coordinator,
+            config_entry.unique_id,
+            TEMPERATURE_SENSORS,
+        )
+        + get_entities(
+            AprilaireStatusSensor, coordinator, config_entry.unique_id, STATUS_SENSORS
+        )
+    )
+
+    async_add_entities(entities)
```

Comment:
Move this up to before the entity classes, and add `from __future__ import annotations` at the top of the module so the type checker allows forward references.

Reply:
Updated